commands/util/binary.js

```diff
@@ -0,0 +1,215 @@
+        }
+
+        const sum = parseInt(first, 2) + parseInt(second, 2);
+        const binarySum = sum.toString(2);
+
+        const embed = new EmbedBuilder()
+          .addFields(
+            { name: "Expression", value: `${first} + ${second}`, inline: true },
+            { name: "Result", value: `${binarySum}`, inline: true }
+          );
+
+        await interaction.reply({ embeds: [embed] });
+
+      } else if (subcommand === "subtract") {
+        const first = interaction.options.getInteger("first").toString();
+        const second = interaction.options.getInteger("second").toString();
+
+        if (!validateBinary(first) || !validateBinary(second)) {
+          throw new Error("Both inputs must be valid binary numbers.");
+        }
+
+        const difference = parseInt(first, 2) - parseInt(second, 2);
+        const binaryDifference = difference.toString(2);
+
+        const embed = new EmbedBuilder()
+          .addFields(
+            { name: "Expression", value: `${first} - ${second}`, inline: true },
+            { name: "Result", value: `${binaryDifference}`, inline: true }
+          );
+
+        await interaction.reply({ embeds: [embed] });
+
+      } else if (subcommand === "multiply") {
+        const first = interaction.options.getInteger("first").toString();
+        const second = interaction.options.getInteger("second").toString();
+
+        if (!validateBinary(first) || !validateBinary(second)) {
+          throw new Error("Both inputs must be valid binary numbers.");
+        }
+
+        const product = parseInt(first, 2) * parseInt(second, 2);
+        const binaryProduct = product.toString(2);
+
+        const embed = new EmbedBuilder()
+          .addFields(
+            { name: "Expression", value: `${first} * ${second}`, inline: true },
+            { name: "Result", value: `${binaryProduct}`, inline: true }
+          );
+
+        await interaction.reply({ embeds: [embed] });
+
+      } else if (subcommand === "divide") {
+        const first = interaction.options.getInteger("first").toString();
+        const second = interaction.options.getInteger("second").toString();
+
+        if (!validateBinary(first) || !validateBinary(second)) {
+          throw new Error("Both inputs must be valid binary numbers.");
+        }
+
+        if (parseInt(second, 2) === 0) {
+          throw new Error("Division by zero is not allowed.");
+        }
+
+        const quotient = Math.floor(parseInt(first, 2) / parseInt(second, 2));
+        const binaryQuotient = quotient.toString(2);
+
+        const embed = new EmbedBuilder()
+          .addFields(
+            { name: "Expression", value: `${first} ÷ ${second}`, inline: true },
+            { name: "Result", value: `${binaryQuotient}`, inline: true }
+          );
+
+        await interaction.reply({ embeds: [embed] });
+      }
+    } catch (error) {
+      await interaction.reply({ content: `An error occurred: ${error.message}`, ephemeral: true });
+    }
```

Comment:
While this does work, it's messy and will display ANY error to the user which we don't want. Just send the user the information they need to care about.

Also see if you can make it into an embed with some error markings (either the color or there's an X to indicate something wrong)